### ISSUE TITLE
`isModuleInstalled` method should return `false` instead of reverting

### ISSUE THREAD

**Issue by cleanunicorn** [erc7579/uMSA#4]
In the current implementation of `isModuleInstalled`, the primary condition is based on `moduleType`. 

https://github.com/erc7579/uMSA/blob/4820e68afdd368077c006dbd876ddd80424c8bcd/src/uMSABasic.sol#L166-L183 

When another contract attempts to verify if a specific module is installed by calling this method, it faces a potential issue. If the module is not installed, the method reverts, necessitating the use of a try/catch pattern (in Solidity or assembly) in the calling contract to handle this scenario.

To simplify interactions with this module, removing the revert in `isModuleInstalled` would be advantageous. If the method returns `false` instead of triggering a revert when a module is not installed, it would encourage developers to use the `supportsModule` method as intended. This approach also eliminates the need for catching a possible revert.

tl;dr;

Do not revert when `moduleType` is unsupported, return `false`.

I am happy to work on this PR if the issue is accepted.

**Comment by zeroknots** [comment]
good catch. was a mistake in uMSABasic. the uMSAAdvanced returned properly. 

fixed here: https://github.com/erc7579/uMSA/commit/4a137b77aa4c128e8d318109a8feb24f503134f9
